src/main/java/com/awesomeorg/airlineservice/service/TicketService.java

```diff
@@ -1,12 +1,81 @@
 package com.awesomeorg.airlineservice.service;
 
+import com.awesomeorg.airlineservice.entity.Ticket;
+import com.awesomeorg.airlineservice.exceptions.TicketAlreadyExistsException;
+import com.awesomeorg.airlineservice.exceptions.TicketNotFoundException;
+import com.awesomeorg.airlineservice.protocol.TicketQuery;
+import com.awesomeorg.airlineservice.protocol.UpdateTicketRequest;
 import com.awesomeorg.airlineservice.repository.TicketRepository;
 import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
 import org.springframework.stereotype.Service;
 
+import java.time.LocalDate;
+import java.util.Optional;
+
 @Service
 @RequiredArgsConstructor
 public class TicketService {
 
     private final TicketRepository ticketRepository;
-}
+
+    public Optional<Ticket> findTicketById(final Long ticketId) {
+        if (ticketId == null) {
+            return Optional.empty();
+        }
+        return ticketRepository.findById(ticketId);
+    }
+
+    public Page<Ticket> findFreeTicket(TicketQuery query, PageRequest pageRequest) {
+        LocalDate dateOfFlight = query.getDateOfFlight();
+        return ticketRepository.findFreeTicket(dateOfFlight, pageRequest);
+    }
+
+
+
+    public Ticket createTicket(TicketQuery request) {
```

Comment:
I don't think that for create you should use `TicketQuery`. Create a `CreateTicketRequest` with all the needed fields for ticket creation.